go/sqlmetrics/connection.go

```diff
@@ -32,8 +32,8 @@ var (
 	connMap sync.Map
 )
 
-func storeConnInfo(conn *proxy.Conn, dsn string) error {
-	info, err := parseDSN(dsn)
+func storeConnInfo(conn *proxy.Conn, driverName, dsn string) error {
+	info, err := parseDSN(driverName, dsn)
```

Comment:
Use dburl package to derive this. Every dsn that connects already has this information. If it doesn't. It may not connect eirher.